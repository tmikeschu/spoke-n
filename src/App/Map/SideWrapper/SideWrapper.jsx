import React from "react"
import PropTypes from "prop-types"
import SuggestionForm from "./SuggestionForm/SuggestionFormContainer"
import Legend from "./Legend/Legend"
import Info from "./Info/Info"
import SuggestionInfo from "./SuggestionInfo/SuggestionInfoContainer"
import Filters from "./Filters/Filters"

const SideWrapper = props => {
  const {
    currentLocation,
    setFilters,
    pinFilters,
    categories
  } = props

  const legend = (
    <Legend
      date={currentLocation && currentLocation.date}
      categories={categories}
    />
  )

  const FiltersBox = (
    <Filters
      categories={categories}
      setFilters={setFilters}
      pinFilters={pinFilters}
    />
  )

  return (
    <section>
      { legend }
      <SuggestionInfo />
      { FiltersBox }
      <div>
        <Info />
        <SuggestionForm />
      </div>
    </section>
  )
}

SideWrapper.propTypes = {
  currentLocation: PropTypes.object.isRequired,
  setFilters: PropTypes.func.isRequired,
  pinFilters: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired
}

export default SideWrapper

