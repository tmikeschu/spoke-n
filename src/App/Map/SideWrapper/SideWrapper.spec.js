import React from "react"
import { shallow } from "enzyme"
import SideWrapper from "./SideWrapper"

const props = {
  currentLocation: { date: "" },
  suggestionInfoIsActive: false,
  setFilters: function() {},
  pinFilters: [],
  categories: [],
  sideClass: "visible",
}

describe("<SideWrapper />", () => {
  const wrapper = shallow(<SideWrapper {...props} />)

  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
