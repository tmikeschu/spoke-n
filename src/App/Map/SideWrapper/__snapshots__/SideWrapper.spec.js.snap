// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideWrapper /> snapshot is valid 1`] = `
Array [
  <section>
    <Legend
        categories={Array []}
        date=""
    />
    <Connect(SuggestionInfo) />
    <Filters
        categories={Array []}
        pinFilters={Array []}
        setFilters={[Function]}
    />
    <div>
        <Info />
        <Connect(SuggestionForm) />
    </div>
</section>,
]
`;
